streams/src/main/java/org/apache/kafka/streams/kstream/SessionWindows.java

```diff
@@ -111,7 +112,7 @@ public SessionWindows until(final long durationMs) throws IllegalArgumentExcepti
             throw new IllegalArgumentException("Window retention time (durationMs) cannot be smaller than window gap.");
         }
 
-        return new SessionWindows(gapMs, durationMs, null);
+        return new SessionWindows(gapMs, durationMs, grace);
```

Comment:
This was actually a bug, which we need to fix regardless of whether we merge this PR or not.